docs/source/running-jobs/environment-variables.rst

```diff
@@ -1,33 +1,67 @@
 
 .. _env-vars:
 
-Using Environment Variables
+Secrets and Environment Variables
 ================================================
 
+Environment variables are a powerful way to pass configuration and secrets to your tasks. There are two types of environment variables in SkyPilot:
+
+- :ref:`User-specified environment variables <user-specified-env-vars>`: Passed by users to tasks, for secrets and arguments.
+- :ref:`SkyPilot environment variables <sky-env-vars>`: Predefined by SkyPilot with information about the cluster and task.
+
+.. _user-specified-env-vars:
+
 User-specified environment variables
 ------------------------------------------------------------------
 
+Passing user-specified environment variables is useful for passing secrets and arguments for ``file_mounts``, ``setup``, and ``run``.
+
 You can specify environment variables to be made available to a task in two ways:
 
-- The ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`
-- The ``--env`` flag in the ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
+- ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`
+
+  .. code-block:: yaml
+
+    envs:
+      MYVAR: val
+  
+- ``--env`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
 
 .. tip::
 
   If an environment variable is required to be specified with `--env` during
-  ``sky launch/exec``, you can set it to ``null`` in task YAML to raise an
+  ``sky launch/exec``, you can set it to empty or ``null`` in task YAML to raise an
   error when it is forgotten to be specified. For example, the ``WANDB_API_KEY``
   and ``HF_TOKEN`` in the following task YAML:
```

Comment:
Can we rephrase like this?

To mark an environment variable as **required** and make SkyPilot forcefully check its existence (errors out if unspecified), set it to an empty string or ``null`` in the task YAML. For example, ``WANDB_API_KEY``
  and ``HF_TOKEN`` in the following task YAML are marked as required:

Reply:
Good point! Updated.